lib/kalibro_client/entities/base.rb

```diff
@@ -14,218 +14,22 @@
-require 'kalibro_client/helpers/hash_converters'
-require 'kalibro_client/helpers/request_methods'
+require 'likeno'
 
 module KalibroClient
   module Entities
-    class Base
-      attr_accessor :kalibro_errors, :persisted
-
-      def initialize(attributes={}, persisted=false)
-        attributes.each { |field, value| send("#{field}=", value) if self.class.is_valid?(field) }
-        @kalibro_errors = []
-        @persisted = persisted
-      end
-
-      def to_hash(options={})
-        hash = Hash.new
-        excepts = options[:except].nil? ? [] : options[:except]
-        excepts << "kalibro_errors"
-        excepts << "persisted"
-        fields.each do |field|
-          hash = field_to_hash(field).merge(hash) if !excepts.include?(field)
-        end
-        hash
-      end
-
-      def self.request(action, params = {}, method = :post, prefix="")
-        response = client.send(method) do |request|
-          url = "/#{endpoint}/#{action}".gsub(":id", params[:id].to_s)
-          url = "/#{prefix}#{url}" unless prefix.empty?
-          request.url url
-          request.body = params unless method == :get || params.empty?
-          request.options.timeout = 300
-          request.options.open_timeout = 300
-        end
-
-        if response.success?
-          response.body
-        # FIXME This condition was added to preserve the deprecated error codes that are returned by kalibro processor
-        elsif response.status == 404 || (response.body.key?('errors') && /NotFound/ === response.body['errors'])
-          raise KalibroClient::Errors::RecordNotFound.new(response: response)
-        else
-          raise KalibroClient::Errors::RequestError.new(response: response)
-        end
-      end
-
-
-      def self.to_object value
-        value.kind_of?(Hash) ? new(value, true) : value
-      end
-
-      def self.to_objects_array value
-        array = value.kind_of?(Array) ? value : [value]
-        array.each.map { |element| to_object(element) }
-      end
-
-      def save
-        if persisted?
-          self.update
-        else
-          without_request_error? do
-            response = self.class.request(save_action, save_params, :post, save_prefix)
-
-            self.id = response[instance_class_name]["id"]
-            self.created_at = response[instance_class_name]["created_at"] unless response[instance_class_name]["created_at"].nil?
-            self.updated_at = response[instance_class_name]["updated_at"] unless response[instance_class_name]["updated_at"].nil?
-            @persisted = true
-          end
-        end
-      end
-
-      def save!
-        return true if save
-        raise KalibroClient::Errors::RecordInvalid.new(self)
+    class Base < Likeno::Entity
+      # We do not track coverage here as it is deprecated and will be removed soon
+      # :nocov:
+      def kalibro_errors
+        STDERR.puts "DEPRECATED: use 'likeno_errors' instead of 'kalibro_errors'."
+        self.likeno_errors
```

Comment:
I think we prefix the errors with `kalibro` and `likeno` i order to avoid conflicts with Rails. But here is a issue to investigate this deeper: https://github.com/mezuro/kalibro_client/issues/106